napari/layers/base/base.py

```diff
@@ -453,26 +451,32 @@ def shear(self, shear):
 
     @property
     def affine(self):
-        """napari.utils.transforms.Affine: Affine transform."""
-        return self._transforms['data2world']
+        """napari.utils.transforms.Affine: Extra affine transform to go from data to world coordinates."""
+        return self._transforms['world2world']
 
     @affine.setter
     def affine(self, affine):
-        if isinstance(affine, np.ndarray) or isinstance(affine, list):
-            self._transforms['data2world'].affine_matrix = np.array(affine)
-        elif isinstance(affine, Affine):
-            affine.name = 'data2world'
-            self._transforms['data2world'] = affine
-        else:
+        self._transforms['world2world'] = self._coerce_affine(affine)
+        self._update_dims()
+        self.events.affine()
+
+    def _coerce_affine(self, affine):
```

Comment:
Can `coerce_affine` be a method in our `layer_utils.py` that also takes an `ndim` kwarg rather than live on the object? I guess it's a little stylistic, but we've tended to prefer having more simple utils functions rather than functions on classes when the function does a simple computation like this

Reply:
Sounds good. Another alternative is in `transform_utils`, where I put some `coerce_*` functions for the transform components like `rotate` in a WIP PR.

Reply:
yup, that sounds like a good place for it

Reply:
Done. I actually put it in `layer_utils` because putting it in `transform_utils` introduced a circular dependency I couldn't easily break.